crates/notary/server/tee/README.md

```diff
@@ -1,3 +1,115 @@
-This folder contains the necessary files to build a Docker image for running the Notary Server on Intel SGX-enabled hardware.
+This folder contains the necessary files to build a Docker image for running the Notary Server on Intel SGX-enabled hardware. 
+
+## Compile the notary server for Intel SGX
+
+We are using [Gramine](https://github.com/gramineproject/gramine) to run the notary server on Intel SGX. Gramine allows to run the notary server in an isolated environment with minimal host requirements.
+
+The isolated environment is defined via the manifest template (`notary-server.manifest.template`).
+
+The notary server for SGX is compiled with an extra Rust feature flag `tee_quote`. This makes the notary server use an ephemeral private notary key for signing attestations (`private_key_pem_path: "/ephemeral/notary.key"`) and also adds the SGX *quote* to the notary server's `/info` end point.
+
+### CI 
 
 The container is built as part of the CI pipeline. For details on the build process, refer to the [CI workflow configuration](../../../../.github/workflows/ci.yml).
+
+CI builds a zip-file `notary-server-sgx.zip` (which contains the compiled binary and the signed manifest). You can find this zip file for all releases and the `dev` builds in the build Artifacts. We also publish a docker image `notary-server-sgx` to <https://github.com/tlsnotary/tlsn/pkgs/container/tlsn%2Fnotary-server-sgx>. Check the #how for more details on running this container.
+
+### Development
+
+You can also build everything locally with `run-gramine-local.sh`. 
+
+This script creates and signs the Gramine manifest for the Notary server in a local development environment. This scripts requires the Gramine SDK, so it is most convenient to run it in a Docker container with the necessary dependencies and tools.
+
+> ⚠️ This script assumes that the `notary-server` binary is already built (for linux/amd64) and available in the current directory. Make sure it is built with the `tee_quote` feature (On Linux: `cargo build --bin notary-server --release --features tee_quote`).
+
+#### Build the Docker Image
+
+To build the Docker image for local development, run:
+```
+docker build -f gramine-local.Dockerfile -t gramine-local .
+```
+#### Run the Gramine Script
+
+Once the image is built, you can run the `run-gramine-local.sh` script inside the container:
+```
+docker run --rm -it \
+  --platform=linux/amd64 \
+  -v "${PWD}:/app" \
+  -w /app/ \
+  gramine-local \
+  "bash -c ./run-gramine-local.sh"
+```
+
+If successful, the script will generate the following files:
+* `notary-server.sig`
+* `notary-server-sigstruct.json`
+* `notary-server.manifest`
+* `notary-server.manifest.sgx`
+
+
+You can verify that the provided **enclave signature (`.sig`)** matches the expected **`MR_ENCLAVE` and `MR_SIGNER`** values, by running the following command inside a **Gramine Docker container** to inspect the enclave's signature:
```

Comment:
provided enclave signature is referring to the notary-server.sig?

Reply:
and `MR_ENCLAVE` and `MR_SIGNER` is from notary-server-sigstruct.json? Is this basically to verify the artefact integrity?